flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java

```diff
@@ -0,0 +1,258 @@
+
+        private final Path location;
+
+        private SchemasSplit(Path location) {
+            this.location = location;
+        }
+
+        @Override
+        public long rowCount() {
+            return new SchemaManager(location).listAllIds().size();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            SchemasSplit that = (SchemasSplit) o;
+            return Objects.equals(location, that.location);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(location);
+        }
+    }
+
+    /** {@link TableRead} implementation for {@link SchemasTable}. */
+    private static class SchemasRead implements TableRead {
+
+        private int[][] projection;
+
+        @Override
+        public TableRead withFilter(Predicate predicate) {
+            return this;
+        }
+
+        @Override
+        public TableRead withProjection(int[][] projection) {
+            this.projection = projection;
+            return this;
+        }
+
+        @Override
+        public RecordReader<RowData> createReader(Split split) throws IOException {
+            if (!(split instanceof SchemasSplit)) {
+                throw new IllegalArgumentException("Unsupported split: " + split.getClass());
+            }
+            Path location = ((SchemasSplit) split).location;
+            Iterator<TableSchema> schemas = new SchemaManager(location).listAll().iterator();
+            Iterator<RowData> rows = Iterators.transform(schemas, this::toRow);
+            if (projection != null) {
+                rows =
+                        Iterators.transform(
+                                rows, row -> ProjectedRowData.from(projection).replaceRow(row));
+            }
+            return new IteratorRecordReader<>(rows);
+        }
+
+        private RowData toRow(TableSchema schema) {
+            List<GenericMapData> fields = new ArrayList<>(schema.fields().size());
+            schema.fields()
+                    .forEach(
+                            f -> {
+                                Map<StringData, StringData> field = new LinkedHashMap<>(4);
+                                field.put(
+                                        StringData.fromString("id"),
+                                        StringData.fromString(String.valueOf(f.id())));
+                                field.put(
+                                        StringData.fromString("name"),
+                                        StringData.fromString(f.name()));
+                                field.put(
+                                        StringData.fromString("type"),
+                                        StringData.fromString(f.type().logicalType().toString()));
```

Comment:
We can use `DataFieldSerializer.typeToJson`. Maybe we can implement `DataType.toString`.

Reply:
Done